SalesforceCommerceCloud/pwa-kit
# [MRT Reference App] @W-13153758 Enable cache for /cache endpoint

# Description

Updates `template-mrt-reference-app/ssr.js` to enable caching when sending a get request to the `/cache` endpoint. This will allow us to write smoke tests for the CloudFrontDistrbution's cache behaviour
https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001Qmt1DYAR/view



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- set response header to `Cache-Control: s-maxage=60` when the endpoint is `/cache`

# How to Test-Drive This PR

- Clone this repo and run `npm ci`
- Start the mrt-reference-app: `cd packages/template-mrt-reference-app && npm start`
- navigate to http://localhost:3000/cache and verify 
  - timestamp is in the response body
  - `"cache-control": "s-maxage=60"` in the response headers
- navigate to any other URL e.g., http://localhost:3000 and verify 
  - timestamp is in the response body
  - `"cache-control": "no-cache"` in the response headers
